src/D2L.CodeStyle.Analyzers/UnsafeStaticsAnalyzer.cs

```diff
@@ -49,6 +49,11 @@ public sealed class UnsafeStaticsAnalyzer : DiagnosticAnalyzer {
         }
 
         private void AnalyzeField( SyntaxNodeAnalysisContext context ) {
+            if( context.Node.SyntaxTree.FilePath.Contains( ".generated" ) ) {

```

Comment:
As chatted about offline: The code does not just exclude things that are in a folder called `.generated`, but also things with a partial name of `.generated`, say `d2l.automation.generated.contract.tests.sln` or similar.